Library/Homebrew/software_spec.rb

```diff
@@ -402,22 +404,38 @@ def compatible_locations?
   # Does the {Bottle} this {BottleSpecification} belongs to need to be relocated?
   sig { returns(T::Boolean) }
   def skip_relocation?
-    cellar == :any_skip_relocation
+    legacy_cellar_or_cellar_for_tag == :any_skip_relocation
   end
 
   def tag?(tag)
     checksum_for(tag) ? true : false
   end
 
-  # Checksum methods in the DSL's bottle block optionally take
+  # BottleChecksum methods in the DSL's bottle block take
   # a Hash, which indicates the platform the checksum applies on.
-  Checksum::TYPES.each do |cksum|
-    define_method(cksum) do |val|
-      digest, tag = val.shift
-      collector[tag] = Checksum.new(cksum, digest)
+  # Example args:
+  # {"22948764d8f8d7be4870ff92dae64d986eb63a9150b219c20fff87d1a6aa93d6"=>
+  # {"tag"=>:mojave, "cellar"=>:any_skip_relocation}}
+  Checksum::TYPES.each do |checksum_type|
+    define_method(checksum_type) do |*args|
+      sha256_line_hash = args.first
+      checksum_hash = sha256_line_hash.select { |key, _value| key.length == 64 }
+      checksum_value = checksum_hash.keys.first
```

Comment:
I don't think `.first` is a good idea here.

Reply:
The hash is of the form: `{"22948764d8f8d7be4870ff92dae64d986eb63a9150b219c20fff87d1a6aa93d6"=>{"tag"=>:mojave, "cellar"=>:any_skip_relocation}}`
I use .select to keep only the key-value with the hexdigest. Then I have a new hash which has only a single key-value, and I take the first one.

If you have an alternative implementation to extract the key/value of a hash without knowing the hash, as it can be a random hexdigest, I am all ears :)